flight4handler.go

```diff
@@ -319,36 +325,28 @@ func flight4Generate(c flightConn, state *State, cache *handshakeCache, cfg *han
 				},
 			})
 		}
-	case cfg.localPSKIdentityHint != nil:
+	case cfg.localPSKIdentityHint != nil || (state.cipherSuite != nil && state.cipherSuite.UsesEllipticCurves()):
```

Comment:
`state.cipherSuite` is asserted earlier (and fatal) so we shouldn't need this check anymore.

Reply:
fixed